src/platforms/alexa/apis/ProactiveEvents.ts

```diff
@@ -0,0 +1,576 @@
+    };
+
+    return this;
+  }
+
+  public setState(status: MESSAGE_ALERT_STATUS, freshness?: MESSAGE_ALERT_FRESHNESS): MessageAlertEventBuilder {
+    this.state = { status, freshness };
+    return this;
+  }
+
+  public build(): MessageAlertEventBuilder {
+    const payload = {
+      messageGroup: this.messageGroup,
+      state: this.state,
+    };
+
+    this.setName("AMAZON.MessageAlert.Activated");
+    this.setPayload(payload);
+
+    return super.build();
+  }
+}
+
+export enum MESSAGE_ALERT_FRESHNESS {
+  NEW = "NEW",
+  OVERDUE = "OVERDUE",
+}
+
+export enum MESSAGE_ALERT_STATUS {
+  FLAGGED = "FLAGGED",
+  UNREAD = "UNREAD",
+}
+
+export enum MESSAGE_ALERT_URGENCY {
+  URGENT = "URGENT",
+}
+
+/**
+ * Order Status Events Builder class reference
+ */
+export class OrderStatusEventBuilder extends EventsBuilder {
+  public state: any = {};
+
+  public setStatus(status: ORDER_STATUS, expectedArrival?: string, enterTimestamp?: string): OrderStatusEventBuilder {
+    this.state = { status, enterTimestamp };
+
+    if (expectedArrival) {
+      this.state.deliveryDetails = { expectedArrival };
+    }
+
+    return this;
+  }
+
+  public build(): OrderStatusEventBuilder {
+    const payload = {
+      order: {
+        seller: {
+          name: "localizedattribute:sellerName",
+        },
+      },
+      state: this.state,
+    };
+
+    this.setName("AMAZON.OrderStatus.Updated");
+    this.setPayload(payload);
+
+    return super.build();
+  }
+}
+
+export enum ORDER_STATUS {
+  ORDER_DELIVERED = "ORDER_DELIVERED",
+  ORDER_OUT_FOR_DELIVERY = "ORDER_OUT_FOR_DELIVERY",
+  ORDER_PREPARING = "ORDER_PREPARING",
+  ORDER_RECEIVED = "ORDER_RECEIVED",
+  ORDER_SHIPPED = "ORDER_SHIPPED",
+  PREORDER_RECEIVED = "PREORDER_RECEIVED",
+}
+
+/**
+ * Occasion Events Builder class reference
+ */
+export class OccasionEventBuilder extends EventsBuilder {
+  public occasion: any = {};
+  public state: any = {};
+
+  public setOccasion(bookingTime: string, occasionType: OCCASION_TYPE): OccasionEventBuilder {
+    this.occasion = {
+      bookingTime,
+      broker: {
+        name: "localizedattribute:brokerName",
+      },
+      occasionType,
+      provider: {
+        name: "localizedattribute:providerName",
+      },
+      subject: "localizedattribute:subject",
+    };
+
+    return this;
+  }
+
+  public setStatus(confirmationStatus: OCCASION_CONFIRMATION_STATUS): OccasionEventBuilder {
+    this.state = { confirmationStatus };
+
+    return this;
+  }
+
+  public build(): OccasionEventBuilder {
+    const payload = {
+      occasion: this.occasion,
+      state: this.state,
+    };
+
+    this.setName("AMAZON.Occasion.Updated");
+    this.setPayload(payload);
+
+    return super.build();
+  }
+}
+
+export enum OCCASION_CONFIRMATION_STATUS {
+  CANCELED = "CANCELED",
+  CONFIRMED = "CONFIRMED",
+  CREATED = "CREATED",
+  REQUESTED = "REQUESTED",
+  RESCHEDULED = "RESCHEDULED",
+  UPDATED = "UPDATED",
+}
+
+export enum OCCASION_TYPE {
```

Comment:
could we take this directly from ask-sdk-model?

Reply:
In the PR's description I mentioned, the ask-sdk-model package in its latest version does not have types for this yet. I did see, they have included the types for Reminders API, so I will implement those changes in my next PR